qiskit/optimization/algorithms/admm_optimizer.py

```diff
@@ -168,15 +168,24 @@ class ADMMOptimizationResult(OptimizationResult):
     """ ADMMOptimization Result."""
 
     def __init__(self, x: Optional[Any] = None, fval: Optional[Any] = None,
-                 state: Optional[ADMMState] = None, results: Optional[Any] = None) -> None:
-        super().__init__(x, fval, results or state)
-        self._state = state
+                 state: Optional[ADMMState] = None, results: Optional[Any] = None,
+                 variables: Optional[Any] = None) -> None:
+        super().__init__(x=x,
+                         variables=variables,
+                         fval=fval,
+                         results=results or state)
+        self.state = state
 
     @property
     def state(self) -> Optional[ADMMState]:
         """ returns state """
         return self._state
 
+    @state.setter
+    def state(self, state: Optional[ADMMState]):
```

Comment:
@adekusar-drl Do you need a setter of state for ADMM optimizer?

Reply:
it is used on line 177,
you can set it explicitly, but I think it is a good practice to use setters

Reply:
Line 177 is the constructor, so it is no problem to set it directly. The question is whether we need to add a public setter or not. If the original does not have a public setter, you should not add it. It will be error-prone.

Reply:
@t-imamichi Imamichi-san, no, there's no need for the setter, it is an internal state of the optimizer, created by the optimizer and kept private by the optimizer. We don't want somebody to alter it. It is only available to a user only after the `solve` method is over.